### ISSUE TITLE
Dump inline frames

### ISSUE THREAD

**Issue by jrmuizel** [mozilla/fix-stacks#15]
I'm not sure how easy this is or whether `symbolic` supports it but it would be nice to have.

As an aside: I was able to get inline frames working for my hacked together https://github.com/jrmuizel/pdb-addr2line but needed to work around https://github.com/willglynn/pdb/issues/78

**Comment by jan-auer** [comment]
Right, `symbolic` does have support for inline frames in PDBs. However, the interface to `DebugSession` might not be super ergonomic, I'm up for suggestions. 

Under the hood, symbolic shells out to `pdb` for all these operations, so we'll have to fix willglynn/pdb#78 first.

**Comment by jrmuizel** [comment]
@mstange what's needed to fix this?

**Comment by mstange** [comment]
I'm not sure why, but fix-stacks currently does a full pre-pass over the debuginfo and [converts it](https://github.com/mozilla/fix-stacks/blob/cfa7048e437c6778b066e7ecb5ce834eb8c3efe7/src/main.rs#L176-L190) into [its own representation](https://github.com/mozilla/fix-stacks/blob/cfa7048e437c6778b066e7ecb5ce834eb8c3efe7/src/main.rs#L112-L113), maybe so that it can query it more efficiently or maybe to save memory.
During the conversion, any inline info is lost.

We could potentially switch it to use a symbolic symcache; those can be queried very efficiently and preserve inline info. However, this would probably break the macOS OSO path, and it has a big startup cost.

Another alternative would be to switch to the Firefox profiler's symbolication infrastructure. I'm currently improving its API so this might become an option soon. The profiler symbolication code can deal with macOS OSOs and it's optimized for fast time-to-first-symbol, i.e. it tries to do only minimal preprocessing.

**Comment by jrmuizel** [comment]
The Firefox profiler's symbolication infrastructure is https://github.com/mstange/samply/tree/main/samply-symbols

**Comment by mstange** [comment]
In the meantime I've polished up "The Firefox profiler's symbolication infrastructure" into a crate called [`wholesym`](https://docs.rs/wholesym).

So the next step here would be to switch fix-stacks to wholesym, which I've attempted in this branch: https://github.com/mstange/fix-stacks/commits/use-wholesym/
